docs/content.zh/release-notes/flink-1.14.md

```diff
@@ -0,0 +1,424 @@
+Moreover, the config options `high-availability.zookeeper.path.latch`
+, `high-availability.zookeeper.path.leader`, `high-availability.zookeeper.path.checkpoints`
+and `high-availability.zookeeper.path.checkpoint-counter` have been removed and, thus, have no
+longer an effect.
+
+#### Fallback value for taskmanager.slot.timeout
+
+##### [FLINK-22002](https://issues.apache.org/jira/browse/FLINK-22002)
+
+The config option `taskmanager.slot.timeout` falls now back to `akka.ask.timeout` if no value has
+been configured.
+
+#### DuplicateJobSubmissionException after JobManager failover
+
+##### [FLINK-21928](https://issues.apache.org/jira/browse/FLINK-21928)
+
+The fix for this problem only works if the ApplicationMode is used with a single job submission and
+if the user code does not access the `JobExecutionResult`. If any of these conditions is violated,
+then Flink cannot guarantee that the whole Flink application is executed.
+
+Additionally, it is still required that the user cleans up the corresponding HA entries for the
+running jobs registry because these entries won't be reliably cleaned up when encountering the
+situation described by FLINK-21928.
+
+#### Zookeeper node under leader and leaderlatch is not deleted after job finished
+
+##### [FLINK-20695](https://issues.apache.org/jira/browse/FLINK-20695)
+
+The `HighAvailabilityServices` have received a new method `cleanupJobData` which can be implemented
+in order to clean up job related HA data after a given job has terminated.
+
+### Checkpoints
+
+#### Change local alignment timeout back to the global time out
+
+##### [FLINK-23041](https://issues.apache.org/jira/browse/FLINK-23041)
+
+The semantic of alignmentTimeout configuration has changed to such meaning:
+
+The time between the start of a checkpoint(on the checkpoint coordinator) and the time when the
+checkpoint barrier is received by a task.
+
+#### Disable unaligned checkpoints for broadcast partitioning
+
+##### [FLINK-22815](https://issues.apache.org/jira/browse/FLINK-22815)
+
+Unaligned checkpoints were disabled for BROADCAST exchanges.
+
+Broadcast partitioning can not work with unaligned checkpointing. There are no guarantees that
+records are consumed at the same rate in all channels. This can result in some tasks applying state
+changes corresponding to a certain broadcasted event while others don't. In turn upon restore, it
+may lead to an inconsistent state.
+
+#### DefaultCompletedCheckpointStore drops unrecoverable checkpoints silently
+
+##### [FLINK-22502](https://issues.apache.org/jira/browse/FLINK-22502)
+
+On recovery, if a failure occurs during retrieval of a checkpoint, the job is restarted (instead of
+skipping the checkpoint in some circumstances). This prevents potential consistency violations.
+
+#### Recover checkpoints when JobMaster gains leadership
+
+##### [FLINK-22483](https://issues.apache.org/jira/browse/FLINK-22483)
```

Comment:
I think we need to rewrite this. No user cares about some internal factory interface.

What they care about is that Flink doesn't go to the external checkpoint storage any more to load checkpoint metadata (except when the JobManager fails over or changes leader), with the effect that the JobManager is much less loaded on failover: Shorter stalls, less external I/O.

Please write this more for the average user, not describing implementation details.

Reply:
@dmvk May I ask you to help me with this?

Reply:
👍

I think we should leave a small note, that this changes the interface as it's public facing (even though 99.99% user won't touch it, we still allow them to override it).

How about:

```
Flink no longer re-loads checkpoint metadata from the external storage before restoring the task state after the failover (except when the JobManager fails over / changes leadership). This results in less external I/O and faster failover.

Please note that this changes a public interfaces around `CompletedCheckpointStore`, that we allow overriding by providing custom implementation of HA Services.

```